src/commands/construct_transaction_command/sign_transaction/sign_with_ledger/mod.rs

```diff
@@ -0,0 +1,334 @@
+use dialoguer::{theme::ColorfulTheme, Input, Select};
+use near_primitives::borsh::BorshSerialize;
+use std::str::FromStr;
+use strum::{EnumDiscriminants, EnumIter, EnumMessage, IntoEnumIterator};
+
+/// Sign constructed transaction with Ledger
+#[derive(Debug, Default, clap::Clap)]
+pub struct CliSignLedger {
+    #[clap(long)]
+    seed_phrase_hd_path: Option<slip10::BIP32Path>,
+    #[clap(subcommand)]
+    submit: Option<Submit>,
+}
+
+#[derive(Debug)]
+pub struct SignLedger {
+    pub seed_phrase_hd_path: slip10::BIP32Path,
+    pub submit: Option<Submit>,
+}
+
+impl From<CliSignLedger> for SignLedger {
+    fn from(item: CliSignLedger) -> Self {
+        let seed_phrase_hd_path = match item.seed_phrase_hd_path {
+            Some(hd_path) => hd_path,
+            None => SignLedger::input_seed_phrase_hd_path(),
+        };
+        let submit: Option<Submit> = item.submit;
+
+        Self {
+            seed_phrase_hd_path,
+            submit,
+        }
+    }
+}
+
+impl SignLedger {
+    fn rpc_client(self, selected_server_url: &str) -> near_jsonrpc_client::JsonRpcClient {
+        near_jsonrpc_client::new_client(&selected_server_url)
+    }
+
+    pub fn input_seed_phrase_hd_path() -> slip10::BIP32Path {
+        let input: String = Input::new()
+            .with_prompt("Enter seed phrase HD Path (if you not sure leave blank for default)")
+            .default("44'/397'/0'/0'/1'".into())
+            .interact_text()
+            .unwrap();
+        slip10::BIP32Path::from_str(input.as_str()).unwrap()
+    }
```

Comment:
```suggestion
    pub fn input_seed_phrase_hd_path() -> slip10::BIP32Path {
        Input::new()
            .with_prompt("Enter seed phrase HD Path (if you not sure leave blank for default)")
            .with_initial_text("44'/397'/0'/0'/1'".into())
            .interact_text()
            .unwrap()
    }
```

See `with_initial_text` and the avoided use of extra `unwrap()` and just directly read to `BIP32Path`